CheatSheetSeriess (#1014)
OWASP/CheatSheetSeries: cheatsheets/AJAX_Security_Cheat_Sheet.md

#### Don't rely on client logic for security

Don't forget that the user controls the client-side logic. A number of browser plugins are available to set breakpoints, skip code, change values, etc. Never rely on client logic for security.

#### Don't rely on client business logic

> Just like the security one, make sure any interesting business rules/logic is duplicated on the server side lest a user bypasses needed logic and does something silly, or worse, costly.

Reviewer: ```suggestion
Just like with security, make sure any important business rules or logic are duplicated on the server side, lest a user bypasses needed logic and does something silly, or worse, costly.
```

#### Avoid building XML or JSON dynamically

> Just like building HTML or SQL you will cause XML injection bugs, so stay away from this or at least use an encoding library or safe JSON or XML library to make attributes and element data safe.

Reviewer: ```suggestion
Just like when building HTML or SQL, you can easily cause XML injection bugs, so stay away from this or at least use an encoding library or safe JSON or XML library to make attributes and element data safe.
```
